ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java

```diff
@@ -108,6 +108,11 @@ public void registerConnection(
             clock,
             permissioningProviders);
     connections.putIfAbsent(peerConnection, peer);
+    // it is common for the peer to be disconnected at this point
+    // if duplicate inbound and outbound connections are established at the same time
+    if (peer.isDisconnected()) {
+      connections.remove(peerConnection);
+    }
```

Comment:
this is the core change here. the rest is logging.

Reply:
This could be a quick fix, but it would be better to sort out the underlying problem wrt how connect / disconnect events are being dispatched.  The comment here isn't really accurate, its actually a bug if these events aren't propagated correctly to `EthPeers` and it doesn't necessarily have to do with duplicate connections. 

On the theory that we're dispatching the disconnect event before the connect event, I haven't looked into this too deeply, but I would probably look at moving [connection dispatch logic](https://github.com/hyperledger/besu/blob/f885c4671333168089a7b86ffedf985fce81954b/ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/RlpxAgent.java#L561) into [`PeerConnectionEvents`](https://github.com/hyperledger/besu/blob/f885c4671333168089a7b86ffedf985fce81954b/ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/connections/PeerConnectionEvents.java#L33).  From there, you could do something like keep a `connectedFuture`, and when [dispatchDisconnect](https://github.com/hyperledger/besu/blob/f885c4671333168089a7b86ffedf985fce81954b/ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/connections/PeerConnectionEvents.java#L50) is invoked, you could chain that logic to `connectedFuture` so that we're ensuring connect and disconnect events are dispatched in order.